distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Table_Helpers.enso

```diff
@@ -143,73 +146,67 @@ type Table_Column_Helper
+            ix : Integer -> if is_index_valid self.internal_columns.length ix then [self.internal_columns.at ix] else
+                problem_builder.report_oob_indices [ix]
+                []
+            Column_Selector.By_Name name case_sensitivity use_regex ->
+                matches = match_columns name case_sensitivity use_regex self.internal_columns
+                if matches.is_empty then problem_builder.report_missing_input_columns [name]
+                matches
+            Column_Selector.Blank_Columns when_any treat_nans_as_blank ->
+                blanks = self.internal_columns.map_with_index ix-> internal_column->
+                    column = self.make_column internal_column
+                    blank_indicator = column.is_blank treat_nans_as_blank
+                    blank_indicator.iif 1 0 . rename "blanks_"+ix.to_text
+
+                ## We cannot just use a custom_column in the aggregate because of
+                   how the column selector works. We may need to revisit this. For
+                   now we need to use tricks like that:
+
+                   To be backend agnostic, we cannot create a new table with the
+                   columns above. Instead, we add our blank columns to the table
+                   and then remove any other columns we. We do not have to deal
+                   with name conflicts, as adding a new column with a clashing
+                   name does not affect the old column or derived columns.
+                table_with_blank_indicators = blanks.fold self.table table-> blanks_col-> table.set blanks_col
+                just_indicators = table_with_blank_indicators.select_columns (blanks.map .name) on_problems=Problem_Behavior.Report_Error
+
+                # Maximum is equivalent to Exists and Minimum is equivalent to Forall.
+                col_aggregate = if when_any then Maximum _ else Minimum _
+                aggregates = blanks.map blanks_col-> col_aggregate blanks_col.name
+
+                aggregate_result = just_indicators.aggregate aggregates on_problems=Problem_Behavior.Report_Error
+                materialized_result = self.materialize <| aggregate_result.catch Any error->
+                    msg = "Unexpected dataflow error has been thrown in an `select_columns_helper`. This is a bug in the Table library. The unexpected error was: "+error.to_display_text
+                    Panic.throw (Illegal_State.Error message=msg cause=error)
+
+                counts = materialized_result.rows.first
+                self.internal_columns.filter_with_index i-> _->
+                    case counts.at i of
+                        ## No rows in input, so treating as blank by convention.
+                        Nothing -> True
+                        1 -> True
+                        0 -> False
+                        _ -> Panic.throw (Illegal_State.Error "Unexpected result. Perhaps an implementation bug of Column_Selector.Blank_Columns.")
```

Comment:
As we discussed, I think the code will be easier to read if we isolte this piece of logic into a separate helper function below.